nixos/modules/services/networking/pounce.nix

```
@@ -0,0 +1,224 @@
+        default = 6697;
+        description = ''
+        '';
+      };
+
+      sslCa = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "~/.config/pounce/pouncecert.pem";
+        description = ''
+          Path for TLS Certificate for CertFP. Needed
+          for logging to Pounce password-less.
+        '';
+      };
+
+      sslKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/private.key";
+        description = ''
+          Path to server TLS certificate key.
+        '';
+      };
+
+      sslCert = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/cert.crt";
+        description = ''
+           Path to server TLS certificate.
+        '';
+      };
+
+      certFP = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/pounce/libera-user.pem";
+        description = ''
+           Path to certFP certificate for login to IRC server
+        '';
+      };
+
+      # If you want to use password, you need to generate it with pounce -x
+      hashedPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+        '';
+      };
+
+      ircHost = mkOption {
+        type = types.str;
+        example = "irc.libera.chat";
+        default = "";
+        description = ''
+          IRC channel
+        '';
+      };
+
+      ircHostPort = mkOption {
+        type = types.int;
+        default = 6697;
+        description = ''
+        '';
+      };
+
+      joinChannels = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          List of channel to join at startup
+        '';
+      };
+
+      ircUser = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC user
+        '';
+      };
+
+      ircNick = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC nickname
+        '';
+      };
+
+      ircPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC password
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "Extra configuration";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.pounce = {
+      description = "Pounce";
+      after = [ "network-online.target" ];
+      wantedBy  = [ "multi-user.target" ];
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.pounce}/bin/pounce ${configFile}";
+        Restart = "always";
+        RuntimeDirectory = "pounce";
+        RuntimeDirectoryMode = "0700";
+      };
+    };
+
+    users.users = optionalAttrs (cfg.user == defaultUser) {
```

```suggestion
    users.users = mkIf (cfg.user == defaultUser) {
```
`optionalAttrs` can get you into trouble with infinite recursion in modules. it's good practice to use `mkIf` instead of the `optional*` family of functions to produce config values unless the default value of `{}` is important.